### PR TITLE
Prevent help popup to disappear when clicking on it

### LINKED ISSUE #65736: Click event badly handled in tip popup in rustdoc
When clicking on the popup, it shouldn't make it disappear!

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #65736.

r? @kinnison

### CODE REVIEW

File: src/librustdoc/html/static/main.js [hunk 1]
```diff
@@ -380,8 +380,22 @@ function getSearchElement() {
...
+            do {
+                x = findParentElement(x, "DIV");
+                if (x) {
+                    x = x.parentNode;
+                }
+                if (x && x.tagName === "ASIDE") {
+                    is_inside_help_popup = true;
+                    break;
+                }
+            } while(x);
```
Comment (kinnison) [inline src/librustdoc/html/static/main.js]: Could this entire loop be replaced with:

```javascript
            var is_inside_help_popup = document.getElementById("help").contains(ev.target);
```

Comment (Dylan-DPC) [reply src/librustdoc/html/static/main.js]: @GuillaumeGomez after making this change, ping me

Comment (GuillaumeGomez) [reply src/librustdoc/html/static/main.js]: @kinnison Awesome! :o

**Comment by GuillaumeGomez** [comment]
Updated!

File: src/librustdoc/html/static/main.js (hunk 1 above)
Comment (kinnison) [reply src/librustdoc/html/static/main.js]: I'd not, I hadn't *tested* that, but I assume @GuillaumeGomez has.

Comment (GuillaumeGomez) [reply src/librustdoc/html/static/main.js]: Actually, I'm wondering about something, let me check it further!

**Comment by GuillaumeGomez** [comment]
Please not merged, I need to check something!

**Comment by GuillaumeGomez** [comment]
So just like I thought, it didn't make the help popup disappear, I updated the code a bit.

**Comment by kinnison** [comment]
Again, I can follow the JS and convince myself it ought to work, but I don't have a test environment right now to actually try it out, so a weak :+1: from me.

**Pull request merged**